CLAUDE.md

```diff
@@ -0,0 +1,152 @@
+# CLAUDE.md
+
+This file provides guidance to Claude Code (claude.ai/code) when working with
+code in the n8n repository.
+
+## Project Overview
+
+n8n is a workflow automation platform written in TypeScript, using a monorepo
+structure managed by pnpm workspaces. It consists of a Node.js backend, Vue.js
+frontend, and extensible node-based workflow engine.
+
+## General Guidelines
+
+- Always use pnpm
+- We use Linear as a ticket tracking system
+- We use Posthog for feature flags
+- When starting to work on a new ticket – create a new branch from fresh
+  master with the name specified in Linear ticket
+- When creating a new branch for a ticket in Linear - use the branch name
+  suggested by linear
+- Use mermaid diagrams in MD files when you need to visualise something
+
+## Essential Commands
+
+### Building
+- `pnpm build` - Build all packages
+- `pnpm build:backend` - Build backend only
+- `pnpm build:frontend` - Build frontend only
+
+ALWAYS redirect the output of the build command to a file:
+```bash
+pnpm build > build.log 2>&1
+```
+
+You can inspect the last few lines of the build log file to check for errors:
+```bash
+tail -n 20 build.log
+```
+
+### Testing
+- `pnpm test` - Run all tests
+- `pnpm test:backend` - Backend tests only
+- `pnpm test:frontend` - Frontend tests only
+- `pnpm dev:e2e` - E2E tests in development mode
+
+Running a particular test file requires going to the directory of that test
+and running: `pnpm test <test-file>`.
+
+When changing directories, use `pushd` to navigate into the directory and
+`popd` to return to the previous directory. When in doubt, use `pwd` to check
+your current directory.
+
+### Code Quality
+- `pnpm lint` - Lint code
```

Comment:
Lint/type check have a dependency on building first, so we should instruct it to do that first.